src/main/java/com/google/alpollo/YouTubeServlet.java

```diff
@@ -0,0 +1,51 @@
+package com.google.alpollo;
+
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.model.SearchListResponse;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/youtube")
+public class YouTubeServlet extends HttpServlet {
+  /**
+   * Accesses YouTube and retrieves a collection of videos relating to the given query/words found
+   * in the request. Returns only the snippet for each video and returns only videos, no playlists
+   * or channels allowed.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+    String query = request.getParameter("query");
+    if (query.equals("")){
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Empty query.");
+      return;
+    }
+
+    YouTube youtubeService = null;
+    try {
+      youtubeService = YouTubeService.getService(ConfigHelper.getSensitiveData(this.getServletContext(), ConfigHelper.SENSITIVE_DATA.API_KEY));
+    } catch (Exception e) {
+      response.getWriter().println(gson.toJson(e));
+      e.printStackTrace();
```

Comment:
I am not sure that is the best way to log the exception here.

Reply:
You mean that it's better to use another methods? For example, something from `java.util.logging`.

Reply:
I just meant that printing the stack trace is not a recommended way of logging exceptions.
As I mentioned in another PR, at some point it would be interesting to figure out proper server-side logging, but that is not a priority.